crates/sui-core/src/unit_tests/transaction_tests.rs

```diff
@@ -978,6 +982,148 @@ async fn test_oversized_txn() {
         .contains("serialized transaction size exceeded maximum"));
 }
 
+async fn get_object_ref(obj_id: &ObjectID, authority_state: &Arc<AuthorityState>) -> ObjectRef {
+    authority_state
+        .get_object(obj_id)
+        .await
+        .unwrap()
+        .unwrap()
+        .compute_object_reference()
+}
+
+// TODO: Next test checks here only good transactions since execution with Random is not yet ready.
+// Testing bad transactions is done in crates/sui-adapter-transactional-tests/tests/a/ptb.move and once execution is
+// ready, we can add tests for good transactions there as well ans remove the next test.
+#[tokio::test]
+async fn test_allowed_ptb_with_random_txn() {
+    telemetry_subscribers::init_for_testing();
+    let (sender, sender_key): (_, AccountKeyPair) = get_key_pair();
+    let num_of_tx = 6;
+    let obj_ids = (0..num_of_tx)
+        .map(|_| ObjectID::random())
+        .collect::<Vec<_>>();
+    let authority_state =
+        init_state_with_ids(obj_ids.iter().map(|id| (sender, *id)).collect::<Vec<_>>()).await;
+    let rgp = authority_state.reference_gas_price_for_testing().unwrap();
+    let initial_shared_version =
+        get_randomness_state_obj_initial_shared_version(authority_state.get_object_store())
+            .unwrap()
+            .unwrap();
+
+    // Publish the package we will call
+    let pkg = build_and_publish_test_package(
+        &authority_state,
+        &sender,
+        &sender_key,
+        &obj_ids[0],
+        "object_basics",
+        /* with_unpublished_deps */ false,
+    )
+    .await
+    .0;
+
+    // Create a server and a client
+    let consensus_address = "/ip4/127.0.0.1/tcp/0/http".parse().unwrap();
+    let server = AuthorityServer::new_for_test(
+        "/ip4/127.0.0.1/tcp/0/http".parse().unwrap(),
+        authority_state.clone(),
+        consensus_address,
+    );
+    let server_handle = server.spawn_for_test().await.unwrap();
+    let client = NetworkAuthorityClient::connect(server_handle.address())
+        .await
+        .unwrap();
+
+    let obj_basics_id = Identifier::new("object_basics").unwrap();
+    let use_clock_id = Identifier::new("use_clock").unwrap();
+    let use_random_id = Identifier::new("use_random").unwrap();
+    let use_value_id = Identifier::new("use_value").unwrap();
```

Comment:
minor nit, bit strange to have `_id` as the suffix for identifiers, since I would have expected these to be objects with that suffix. 
Probably fine to just drop the `_id` suffix